### ISSUE TITLE
`prevent-abbreviations` should check all-uppercase variable names

### ISSUE THREAD

**Issue by ehmicky** [sindresorhus/eslint-plugin-unicorn#269]
`a.js`:

```js
const opts = []
```

```bash
$ eslint a.js
/home/user/a.js
  1:7  error  The variable `opts` should be named `options`. A more descriptive name will do too  unicorn/prevent-abbreviations

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

`b.js`:

```js
const OPTS = []
```

```bash
$ eslint b.js
```

`b.js` does not report any error but it should. 

# Configuration

`.eslintrc.json`:

```json
{
  "parserOptions": {
    "ecmaVersion": 2019
  },
  "plugins": ["unicorn"],
  "rules": {
    "unicorn/prevent-abbreviations": "error"
  }
}
```

# Environment

```bash
$ npm ls eslint eslint-plugin-unicorn
desktop@1.0.0 /home/user/Desktop
├── eslint@5.15.3 
└── eslint-plugin-unicorn@8.0.0

$ node --version
v11.12.0

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.10
Release:	18.10
Codename:	cosmic
```

**Comment by sindresorhus** [comment]
It was an intentional choice to ignore those:
- https://github.com/sindresorhus/eslint-plugin-unicorn/pull/237#issuecomment-470938908
- https://github.com/sindresorhus/eslint-plugin-unicorn/pull/237/commits/75035550e37c169671eaef876e1162676addb27b

**Comment by ehmicky** [comment]
Reading up the comment, it seems that the reason was to prevent reporting environment variables, since they are almost always uppercase, and user might not decide on them considering they come from the calling process.

This makes sense.

However, it's too bad because some users (like me) like to name constants with uppercase letters. Those variables cannot be linted. Is there a way this could be done without getting issues with environment variables?

**Comment by fregante** [comment]
Closing for now, no movement in 3+ years
